ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/reindexfromstate/StateToUpdateMapping.scala

```diff
@@ -0,0 +1,277 @@
+        )
+
+      case DamlStateKey.KeyCase.CONFIGURATION =>
+        val configurationEntry = damlStateValue.getConfigurationEntry
+        Configuration.decode(configurationEntry.getConfiguration).toOption.map { config =>
+          ConfigurationChanged(
+            recordTime(),
+            SubmissionId.assertFromString(configurationEntry.getSubmissionId),
+            ParticipantId.assertFromString(configurationEntry.getParticipantId),
+            config,
+          )
+        }
+
+      case DamlStateKey.KeyCase.PARTY =>
+        val party = damlStateValue.getParty
+        Some(
+          PartyAddedToParticipant(
+            Ref.Party.assertFromString(damlStateKey.getParty),
+            party.getDisplayName,
+            ParticipantId.assertFromString(party.getParticipantId),
+            recordTime(),
+            None,
+          )
+        )
+
+      case DamlStateKey.KeyCase.CONTRACT_KEY =>
+        // Handled as part of CONTRACT_ID.
+        None
+
+      case DamlStateKey.KeyCase.CONTRACT_ID =>
+        val contract = damlStateValue.getContractState
+        if (contract.hasArchivedAt) {
+          println(s"Skipping archived contract ${contract.getContractKey}")
+          None
+        } else {
+          Some(createTransactionAcceptedForContract(damlStateKey.getContractId, contract))
+        }
+
+      case DamlStateKey.KeyCase.COMMAND_DEDUP | DamlStateKey.KeyCase.SUBMISSION_DEDUP =>
+        None
+
+      case DamlStateKey.KeyCase.KEY_NOT_SET =>
+        None
+    }
+  }
+
+  private val engine = new Engine()
+
+  private def createTransactionAcceptedForContract(
+      contractId: String,
+      contract: DamlContractState,
+  ): TransactionAccepted = {
+    val contractInstance = Conversions.decodeContractInstance(contract.getContractInstance)
+    val ledgerEffectiveTime = Timestamp.assertFromInstant(
+      Instant.ofEpochSecond(
+        contract.getActiveAt.getSeconds,
+        contract.getActiveAt.getNanos.toLong,
+      )
+    )
+    val submissionSeed = crypto.Hash.assertHashContractInstance(
+      contractInstance.template,
+      contractInstance.arg.value,
+    )
+
+    TransactionAccepted(
+      optSubmitterInfo = None,
+      transactionMeta = TransactionMeta(
+        ledgerEffectiveTime = ledgerEffectiveTime,
+        workflowId = None,
+        submissionTime = ledgerEffectiveTime,
+        submissionSeed = submissionSeed,
+        optUsedPackages = None,
+        optNodeSeeds = None,
+        optByKeyNodes = None,
+      ),
+      transaction = transactionFromContractState(
+        contractId,
+        contract,
+        contractInstance,
+        submissionSeed,
+      ),
+      transactionId = TransactionId.assertFromString(
+        UUID.randomUUID().toString
+      ), // TODO: Can be determined from state?
+      recordTime = recordTime(),
+      divulgedContracts = List.empty, // No divulged contracts at creation.
+      blindingInfo = Some(blindingInfoFromContractState(contract, contractId)),
```

Comment:
@miklos-da : I believe you can save code by setting this to `None`, as the indexer will then compute it by itself from the `CommittedTransaction`